x-pack/plugins/apm/server/lib/service_nodes/index.ts

```diff
@@ -68,15 +68,18 @@ const getServiceNodes = async ({
     return [];
   }
 
-  return response.aggregations.nodes.buckets.map((bucket) => {
-    return {
+  return response.aggregations.nodes.buckets
+    .map((bucket) => ({
       name: bucket.key as string,
       cpu: bucket.cpu.value,
       heapMemory: bucket.heapMemory.value,
       nonHeapMemory: bucket.nonHeapMemory.value,
       threadCount: bucket.threadCount.value,
-    };
-  });
+    }))
+    .filter(
+      (item) =>
+        item.cpu || item.heapMemory || item.nonHeapMemory || item.threadCount
```

Comment:
I think it's good if we are a little bit more specific here, and check for `null`, depending on what kind of aggregation it is. Some aggregations will return `null` when there is no data, others will report 0 (e.g. a cardinality agg), but in most cases, there is data if the value is 0, which is what we are eliminating here.